bin/schema_generator.sh

```diff
@@ -56,124 +52,72 @@ function setup_git_ssh() {
+    pushd .
 
-    [[ -d mozilla-pipeline-schemas ]] && rm -r mozilla-pipeline-schemas
-
+    [[ -d mozilla-pipeline-schemas ]] && rm -rf mozilla-pipeline-schemas
     git clone "$MPS_REPO_URL"
-    cd mozilla-pipeline-schemas/$MPS_SCHEMAS_DIR
+    cd mozilla-pipeline-schemas
+    git fetch --all
+    git checkout "$MPS_BRANCH_PUBLISH"
     git checkout "$MPS_BRANCH_SOURCE"
-    git checkout -b $MPS_BRANCH_WORKING
-}
 
-function filter_schemas() {
-    # Remove BigQuery schemas that are in the disallow list
-    find . -name '*.bq' | grep -f $DISALLOWLIST | xargs rm -f
+    popd
 }
 
-function create_changelog() {
-    # Generate output for referencing to the changeset in the source branch.
-    # Useful references for --format and --pretty
-    # https://stackoverflow.com/questions/25563455/how-do-i-get-last-commit-date-from-git-repository
-    # https://stackoverflow.com/questions/1441010/the-shortest-possible-output-from-git-log-containing-author-and-date
-    # https://git-scm.com/docs/git-log/1.8.0#git-log---daterelativelocaldefaultisorfcshortraw
-    local start_date
-    # NOTES: Since this function is looking at commit dates (such as rebases),
-    # it's best to enforce squash/rebase commits onto the master branch. If this
-    # isn't enforced, it's possible to miss out on certain commits to the
-    # generated branch. Another solution is to generate a tag on master before
-    # the generator is run. However, the heuristic of using the latest commit
-    # date works well enough.
-    start_date=$(git log "${MPS_BRANCH_PUBLISH}" -1 --format=%cd --date=iso)
-    git log "${MPS_BRANCH_SOURCE}" \
-        --since="$start_date" \
-        --pretty=format:"%h%x09%cd%x09%s" \
-        --date=iso
+function validate_bigquery {
+    python3 "${BIN}/validate_bigquery" "$@"
 }
 
-function commit_schemas() {
-    # This method will keep a changelog of releases. If we delete and newly
-    # checkout branches everytime, that will contain a changelog of changes.
-    # Assumes the current directory is the root of the repository.
-    # Note that in this step we throw out all the temporary modifications we've
-    # made to the JSON schemas.
-    find . -name "*.bq" -type f -exec git add {} +
-    git checkout ./*.schema.json
-
-    # Add Glean JSON schemas with generic schema
-    mozilla-schema-generator generate-glean-pings --out-dir $MPS_SCHEMAS_DIR --pretty --generic-schema --mps-branch $MPS_BRANCH_SOURCE
-    find . -name "*.schema.json" -type f -exec git add {} +
-
-    git commit -a -m "Interim Commit"
-
-    git checkout "$MPS_BRANCH_PUBLISH" || git checkout -b "$MPS_BRANCH_PUBLISH"
-
-    # Keep only the schemas dir
-    find . -mindepth 1 -maxdepth 1 -not -name .git -exec rm -rf {} +
-    git checkout $MPS_BRANCH_WORKING -- schemas
-    git commit --all \
-        --message "Auto-push from schema generation [ci skip]" \
-        --message "$(create_changelog)" \
-        || echo "Nothing to commit"
+function validate_generate_commit() {
```

Comment:
```suggestion
function validate_and_generate_commit() {
```

This seems a bit clearer in describing what this procedure is doing

Reply:
I personally don't like adding in the conjunction, since it makes the function name harder to remember. It's the same reason why I renamed `setup_git_ssh` and `clone_and_configure_mps` to `setup_git` and `setup_mps` respectively. 

It could be interpreted as "validating generated commits" instead of "generate commit and validate the output", though.

Reply:
> It could be interpreted as "validating generated commits" instead of "generate commit and validate the output", though.

I agree the conjunction is uglier, but this is why the name seemed confusing.